contrib/verifybinaries/test.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+import json
+import sys
+import subprocess
+from pathlib import Path
+
+
+def main():
+    """Tests ordered roughly from faster to slower."""
+
+    expect_code(run_verify('0.32'), 4, "Nonexistent version should fail")
+    expect_code(run_verify('0.32.awefa.12f9h'), 11, "Malformed version should fail")
+    expect_code(run_verify('22.0 --min-trusted-sigs 20'), 9, "--min-trusted-sigs 20 should fail")
+
+    _220 = run_verify('22.0 --json')
+    result = json.loads(_220.stdout.decode())
+
+    print("- testing multisig verification (22.0)", flush=True)
+    expect_code(_220, 0, "22.0 should succeed")
+    v = result['verified_binaries']
+    assert result['good_trusted_sigs']
+    assert v['bitcoin-22.0-aarch64-linux-gnu.tar.gz'] == 'ac718fed08570a81b3587587872ad85a25173afa5f9fbbd0c03ba4d1714cfa3e'
+    assert v['bitcoin-22.0-osx64.tar.gz'] == '2744d199c3343b2d94faffdfb2c94d75a630ba27301a70e47b0ad30a7e0155e9'
+    assert v['bitcoin-22.0-x86_64-linux-gnu.tar.gz'] == '59ebd25dd82a51638b7a6bb914586201e67db67b919b2a1ff08925a7936d1b16'
+
+    print("- testing single-sig verification (0.20.0)", flush=True)
+    _20 = run_verify('0.20.0 --json')
+    result = json.loads(_20.stdout.decode())
+    v = result['verified_binaries']
+
+    expect_code(_20, 0, "0.20.0 should succeed")
+    assert v['bitcoin-0.20.0-aarch64-linux-gnu.tar.gz'] == '081b30b0f1af95656242c83eef30bbf7216b1a30fa8e8f29b3b160fe520d28f6'
+    assert v['bitcoin-0.20.0-arm-linux-gnueabihf.tar.gz'] == '05014c7ff00f4496b1f389f0961d807e04505d8721d5c6f69567f2a0ec1985cc'
+    assert v['bitcoin-0.20.0-osx64.tar.gz'] == '34f377fee2c7adf59981dde7e41215765d47b466f773cf2673137d30495b2675'
+
+
+def run_verify(extra: str) -> subprocess.CompletedProcess:
+    maybe_here = Path.cwd() / 'verify.py'
+    path = maybe_here if maybe_here.exists() else Path.cwd() / 'contrib' / 'verifybinaries' / 'verify.py'
+
+    return subprocess.run(
+        f"{path} --noninteractive --cleanup {extra}", capture_output=True, shell=True)
```

Comment:
Capture_output is python 3.7. Can just capture both with `subprocess.PIPE` though:

```suggestion
        f"{path} --noninteractive --cleanup {extra}", stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
```